docker-storage-setup.sh

```diff
@@ -707,32 +707,56 @@ scan_disks() {
 }
 
 create_partition_sfdisk(){
-  local dev="$1" size
+  local dev="$1" size label="$2" type
   # Use a single partition of a whole device
   # TODO:
-  #   * Consider gpt, or unpartitioned volumes
+  #   * Consider unpartitioned volumes
   #   * Error handling when partition(s) already exist
   #   * Deal with loop/nbd device names. See growpart code
+
   size=$(( $( awk "\$4 ~ /"$( basename $dev )"/ { print \$3 }" /proc/partitions ) * 2 - 2048 ))
+  # if parted is not present, and this version of sfdisk
+  # does not support GPT then we are limited to 2TB
+  if [ $label = "gpt" ]; then
+    if $(sfdisk --help|grep gpt >/dev/null) ; then
+      label='label: gpt'
+      type=''
+      size=$(( $size - 34 ))
+    else
+      Info "Your version of sfdisk does not support GPT labels. Will use MBR instead but partition size will be limited to 2TB"
+      label=''
+      type=', Id=8e'
+      # 2TB in sectors - 2048
+      if [ $size -gt 4294965248 ]; then size=4294965248; fi
+    fi
+  elif [ $label = "msdos" ]; then
+    label=''
+  fi
+  
     cat <<EOF | sfdisk $dev
 unit: sectors
+$label
 
-${dev}1 : start=     2048, size=  ${size}, Id=8e
+${dev}1 : start=     2048, size=  ${size}${type}
 EOF
 }
 
 create_partition_parted(){
-  local dev="$1"
-  parted $dev --script mklabel msdos mkpart primary 0% 100% set 1 lvm on
+  local dev="$1" label="$2"
+  parted $dev --script mklabel $label mkpart primary 0% 100% set 1 lvm on
```

Comment:
Oh, I didn't know that. Interestingly parted doesn't say anything in interactive mode, but it does on the man pages:

```
(parted) help mkpart                                                           
  mkpart PART-TYPE [FS-TYPE] START END     make a partition

	PART-TYPE is one of: primary, logical, extended
        FS-TYPE is one of: btrfs, nilfs2, ext4, ext3, ext2, fat32, fat16, hfsx, hfs+, hfs, jfs, swsusp, linux-swap(v1), linux-swap(v0), ntfs, reiserfs, hp-ufs, sun-ufs, xfs, apfs2, apfs1, asfs, amufs5, amufs4,
        amufs3, amufs2, amufs1, amufs0, amufs, affs7, affs6, affs5, affs4, affs3, affs2, affs1, affs0, linux-swap, linux-swap(new), linux-swap(old)
        START and END are disk locations, such as 4GB or 10%.  Negative values count from the end of the disk.  For example, -1s specifies exactly the last sector.
        
        'mkpart' makes a partition without creating a new file system on the partition.  FS-TYPE may be specified to set an appropriate partition ID.

# man parted
(...)
              mkpart [part-type name fs-type] start end
                     Create  a  new partition. part-type may be specified only with msdos and dvh partition tables, it should be one of "primary", "logical", or "extended".  name is required for GPT partition tables and fs-type is optional.  fs-type can be one of "btrfs", "ext2", "ext3", "ext4", "fat16", "fat32", "hfs", "hfs+", "linux-swap", "ntfs", "reiserfs", or "xfs".
```

So yes, the above would set "primary" as the partition name. Since a name is required, do you have any preferences for which name to use?